pkg/runtime/scheduler/clients/clients.go

```diff
@@ -62,6 +62,12 @@ func New(ctx context.Context, opts Options) (*Clients, error) {
 
 // Next returns the next client in a round-robin manner.
 func (c *Clients) Next() schedulerv1pb.SchedulerClient {
+	// This is called even when no scheduler is initialized.
+	// Keeping these nil checks here as a central location instead of spreading "ifs".
+	if c == nil || c.clients == nil || len(c.clients) == 0 {
+		return &schedulerClientNotInitializedInstance
```

Comment:
Does it make more sense to not return an error here instead? Or better, also pass a context whereby it waits for client to become available. We return the context error when it has been cancelled.

Reply:
I am keeping this change minimal for the sake of this release.

Reply:
If you want to make another PR in master improving this, I am all in. For 1.14, I recommend us being minimalistic.